### PR TITLE
Update for Welcome Message /overhaul

### PR DESCRIPTION
**Pull request by Bacon-Fixation**
***Notice: this PR has been rechecked manually for Commit issues I've been having and these file changes are what I intended***

Files Updated 12/18/2020 i apologize i went ham and reduce the DB calls along with the original request

As requested from #361 
an ability to have the message posted into a desired text channel with a ping (default is still dm's)
image

- Tested on
Windows 10 (local host)

Much Love
-Bacon

### CODE REVIEW

**Comment by Bacon-Fixation** [comment]
Heads up I just found [discord-canvas](https://www.npmjs.com/package/discord-canvas) and currently looking into cleaning up the Welcome message with better setting controls using this addition I think it would make it possible for easier configuring through the command with possible better end results. 

And possibly easier to read if modification are needed to the file or command

**Comment by Bacon-Fixation** [comment]
the update does require the guild to rerun their welcome setting command

**Comment by Bacon-Fixation** [comment]
fixed error that can occur if the user just updates master-bot and doesn't have the new addition in the guilds db data

**Comment by galnir** [comment]
What happens if the user's DMs are private? I remember it used to throw errors when such a scenario happened

**Comment by Bacon-Fixation** [comment]
```
    if (
      welcomeMsgSettings.destination == 'direct message' ||
      !welcomeMsgSettings.destination
    )
      try {
        await member.user.send(embed);
      } catch {
        console.log(`${member.user.username}'s dms are private`);
      }
```

**Comment by galnir** [comment]
> it should still goto console like it used to
> 
> ```
>     if (
>       welcomeMsgSettings.destination == 'direct message' ||
>       !welcomeMsgSettings.destination
>     )
>       try {
>         await member.user.send(embed);
>       } catch {
>         console.log(`${member.user.username}'s dms are private`);
>       }
> ```

right, didn't see that. Is the feature done?

**Comment by Bacon-Fixation** [comment]
yep yep

**Comment by galnir** [comment]
You're making wayy to many database calls here. It's not a good idea at all to call the database many times when you can just call it once and get all data you need by accessing the first value.
For example instead of getting `${message.member.guild.id}.serverSettings.welcomeMsg.topImageText`, just get `${message.member.guild.id}.serverSettings` and access the properties you need via the variable you stored the value at.
The same goes for setting, it's not considered good practice to set each property with its own db call, just set the properties on a variable and then set the variable with a single call to the database

**Comment by Bacon-Fixation** [comment]
Ok Sweet TY for the info I will go back and do better :D

File: commands/guild/welcome-message-settings.js [hunk 1]
```diff
@@ -102,157 +113,105 @@ module.exports = class WecomeSettingsCommand extends Command {
...
-    db.set(
-      `${message.member.guild.id}.serverSettings.welcomeMsg.cmdUsed`,
-      message.content
+      }),
+      timeStamp: message.createdAt,
+      cmdUsed: message.content
+    });
+    const grabDB = db.get(
+      `${message.member.guild.id}.serverSettings.welcomeMsg`
     );
```
Comment (Bacon-Fixation) [inline commands/guild/welcome-message-settings.js:190]: @galnir So something like this ?

**Comment by Bacon-Fixation** [comment]
..... English is hard work man!!! (only verbal language i know and I can't spell to save my life OMG just don't look at the commit descriptions ...... sorry not sorry kinda thing :'( i can't edit them)

**Comment by Bacon-Fixation** [comment]
I believe I understood what you meant and feel the feature ready for review

**Comment by Bacon-Fixation** [comment]
I believe I understood what you meant and feel the feature/update ready for review 

-Count is now 
`welcome-message-settings.js` 2 
`welcome-message.js`  ("no" is 1)  ("yes" 2 if no DB present or 1 with a DB)
`index.js` 1


Much Love
-Bacon

**Comment by Bacon-Fixation** [comment]
I apologize
i don't know why that did't go with the others,  it was still in stash

**Comment by Bacon-Fixation** [comment]
I apologize 
I don't know why that didn't go with the others when I committed to PR an ~hour ago,  it was still in stash

**Comment by Bacon-Fixation** [comment]
Finished testing this morning

Much Love
-Bacon

**Comment by Bacon-Fixation** [comment]
@galnir forgive me for Pinging and asking here, but I noticed that `ffmpeg-static@2.70` is out dated and that there is a `ffmpeg-static@4.3.1` just curious if there was a reason for not updating  to the [current node version ](https://www.npmjs.com/package/ffmpeg-static)

Much Love
-Bacon

**Comment by Bacon-Fixation** [comment]
Forgive me galnir for the needless ping I deleted my question I found the answer why ffmpeg-static is the older 2.7.0 version instead of a newer one

Much Love
-Bacon

**Comment by galnir** [comment]
> Edit
> 
> Original Question
> 
> ```
> @ galnir forgive me for Pinging and asking here, but I noticed that ffmpeg-static@2.70 is out dated and
> that there is a ffmpeg-static@4.2.7 just curious if there was a reason for not updating to the current node version
> 
> at a glance it seems like there was a reason why it was never updated being one of the oldest version 
> in Master-Bot Dep list) when there's an update available
> 
> Edit: just tried ffmpeg-static@4.2.7 works for me but I'm not sure how to test this thoroughly
> 
> Noticed when looking into a personal issue with music broadcast stopping regularly after 3 hours of continuous use ,
> queue is full but no sound, requires !leave and redo my music queue to bring it back,
>  no errors occur when broadcast just stops all of a sudden (even while a song was playing and still had time left
> ```
> 
> Forgive me galnir for the needless ping, I came across the answer why ffmpeg-static is the older 2.7.0 version instead of a newer one.
> It requires the ffmpeg install to be the same or higher than the Node Dep `ffmpeg-static` version would cause a confusing error in console in some situations(noticed on Arm6 build Only)
> 
> Much Love
> -Bacon

for some reason it wouldn't update it unless you add ```@latest``` at the end of the name of the npm package. Anyway I updated both ytdl-core and ffmpeg-static in ddc5634

**Comment by Bacon-Fixation** [comment]
Couldn't find binaries to build from

**Comment by Bacon-Fixation** [comment]
Closing to Re-Do commit with other account Just in Case

**Pull request closed without merging**